src/main/java/com/haxwell/apps/quizki/services/QuestionServiceImpl.java

```diff
@@ -247,4 +253,48 @@ public CreatedQuestionDTO getQuestionById(String idString) throws GetQuestionExc
 		
 	}
 	
+	
+	public ArrayList<CreatedQuestionDTO> getQuestions(int page, int size) throws GetQuestionException {
+	
+		Pageable pageable = null;
+		
+		outputDTOs = new ArrayList<CreatedQuestionDTO>();
+		
+		long qs = questionRepo.count();
+		
+		if((page - 1) * size <= qs) {
+			
+			pageable = PageRequest.of((page - 1), size);
+			
+		} else {
+			return outputDTOs;		//return the empty array if no data exists
+		}
+			
+		questions = questionRepo.findAll(pageable);
+		
+		outputDTOs.clear();
```

Comment:
Just for good measure? ;)

Reply:
Not sure what you're referring to here, the clear statement?

Reply:
Kinda, `ouputDTOs` is class level and would retain the data from previous tests.